### DISCUSSION TITLE
Github action won't start

### DISCUSSION THREAD

**Discussion by aq-mori-yoshioka** [community/community#46633] [Actions] [answered]
### Select Topic Area

Question

### Body

Hello.

I am trying to configure github action.
What I want to achieve is souce code mirroring from github repository to AWS codecommit repository,
it triggered when I push updated code to github repository.

yaml file in workflow is as below. 

/workflows/main.yaml
- - - - - - - - - - - - - - - - - - - - - - - - - - - 
name: Mirroring

on: 
  push:
    branches: 
      - deploy
  workflow_dispatch:

jobs:
  print-workflow-started:
    runs-on: ubuntu-latest
    steps:
      - name: execute echo command
        run: echo "Workflow started"
  to_codecommit:
    runs-on: ubuntu-18.04
    steps:
      - uses: actions/checkout@v1
        with:
          fetch-depth: 0
      - uses: pixta-dev/repository-mirroring-action@v1
        with:
          target_repo_url:
            ssh://git-codecommit.ap-northeast-1.amazonaws.com/v1/repos/*************
          ssh_private_key:
            ${{ secrets.CODECOMMIT_SSH_PRIVATE_KEY }}
          ssh_username:
            ${{ secrets.CODECOMMIT_SSH_PRIVATE_KEY_ID }}
- - - - - - - - - - - - - - - - - - - - - - - - - - - 

Default branch is "develop" and I'm testing with deploy branch.

I exucuted git push several times and succeeded but no action occurred on github.
No error message or log exists so I don't know how to solve it.

I think I've missed some basic setting.
Could you please review my configuration and tell me how to achive my requirements?

Thanks.

**Comment by spenserblack** [comment, marked as answer]
> /workflows/main.yaml

Workflows should be in `.github/workflows/<filename>.yml`

Also, does the workflow exist in the `deploy` branch?

**Reply by aq-mori-yoshioka** [reply]
spenserblack, thank you for your reply.

> Workflows should be in .github/workflows/<filename>.yml

Thank you very much!
I put workflows under .github and pushed then workflow started.

> Also, does the workflow exist in the deploy branch?

Yes. workflows directory only exists in deploy branch.


While github action (in jobs: to_codecommit: ) I encoutered new error.
- - - - - - - - - - - - - - - - - - - - 
error: refusing to delete the current default branch 'refs/heads/master'.
- - - - - - - - - - - - - - - - - - - - 

Mirroring itself succeed and codecommit repository's default branch is "master".
How can I fix it or is it ignorable?


Thank you.

**Reply by spenserblack** [reply]
That's probably raised by https://github.com/pixta-dev/repository-mirroring-action. I'm not familiar with that action, but I'm guessing that it tries to delete branches that exist in the target but don't exist in the source when mirroring. I don't think that there is a single "correct" way to handle this. It depends on how you want to set up the target repository. Perhaps it's possible to remove protections from `master` in the target repository?

**Reply by aq-mori-yoshioka** [reply]
I set main branch as default and deleted master branch on codecommit repository.
After that codecommit mirroring succeed.

Thanks for your prompt help.
